node/src/accountant/scanners/mod.rs

```diff
@@ -854,24 +866,31 @@ impl Scanner<RetrieveTransactions, ReceivedPayments> for ReceivableScanner {
     }
 
     fn finish_scan(&mut self, msg: ReceivedPayments, logger: &Logger) -> Option<NodeToUiMessage> {
-        if msg.payments.is_empty() {
-            info!(
-                logger,
-                "No newly received payments were detected during the scanning process."
-            );
-
-            match self
-                .persistent_configuration
-                .set_start_block(msg.new_start_block)
-            {
-                Ok(()) => debug!(logger, "Start block updated to {}", msg.new_start_block),
-                Err(e) => panic!(
-                    "Attempt to set new start block to {} failed due to: {:?}",
-                    msg.new_start_block, e
-                ),
+        match msg.scan_result {
+            Ok(payments_and_start_block) => {
+                self.handle_new_received_payments(&payments_and_start_block, msg.timestamp, logger);
             }
-        } else {
-            self.handle_new_received_payments(&msg, logger)
+            Err(e) => match e {
+                ReceivedPaymentsError::ExceededBlockScanLimit(max_block_count) => {
+                    debug!(logger, "Writing max_block_count({})", max_block_count);
+                    self.persistent_configuration
+                        .set_max_block_count(Some(max_block_count))
```

Comment:
The Block count is fully updated and tested. This was part of a new design.

Reply:
I understand that this works but it is conceptually wrong. That is from the standpoint of "separate responsibilities", which we want to be serious about even in small units, but the more in the terms of an entire actor and its set of responsibilities. This change you made drift us away from cleanly defined competence of the Accountant. 

Again, in the ideal world, this would require keeping this actor clean of anything that refers to the blockchain technology. We've concluded in the past that it cannot be done entirely so, or it would be significantly hard. More was said in my previous comment already.

The problem with your new change is that it could easily have been handled in the former location, the BlockchainBridge, which is the Actor which should be looking after all the things with the Blockchain, and so also setting `set_max_block_count`.

In my eyes, this could have an actual adverse impact in a long term. Please, at least bring this up with Dan. He should be who decides these quite top-level architectural things.